### PR TITLE
Categories CRUD and views without design

### PR DESCRIPTION
**Pull request by JoseRicardoJimenez**
i create Edit, New and Index views
i add categories controller with CRUD functions

# Description
What did you implemented/Why did you implemented this:

You can create, update and delete any category you need

### CODE REVIEW

File: local_sales/app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  def edit
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      flash[:notice] = 'Category was successfully created'
+      redirect_to categories_path
+    else
+      flash[:notice] = 'Category was not created'
```
Comment (juliocabrera820) [inline local_sales/app/controllers/categories_controller.rb:21]: You could add the option of rendering new view
```ruby
render :new
```

File: local_sales/app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    end
+  end  
+
+  def update
+    if @category.update(category_params)
+      flash[:notice] = 'Category was successfully updated'
+      redirect_to categories_path
+    else
+      flash[:notice] = 'Category was not updated'
```
Comment (juliocabrera820) [inline local_sales/app/controllers/categories_controller.rb:30]: Same here, render edit view

**Review by juliocabrera820 — APPROVED** [review]
LGTM

File: local_sales/app/controllers/categories_controller.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+class CategoriesController < ApplicationController
+  before_action :set_product, only: %i[show edit update destroy]
```
Comment (juliocabrera820) [inline local_sales/app/controllers/categories_controller.rb:2]: this won't work, the method must be *set_product* as you defined at your private methods

File: local_sales/app/controllers/categories_controller.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      redirect_to categories_path
+    else
+      flash[:notice] = 'Category was not updated'
+    end
+  end
+
+  def destroy
+    @category.destroy
+    flash[:notice] = 'Category was succesfully destroyed'
+    redirect_to @company
```
Comment (juliocabrera820) [inline local_sales/app/controllers/categories_controller.rb:37]: You must redirect to *companies_path*

**Review by juliocabrera820 — CHANGES_REQUESTED** [review]
You can add some actions

File: local_sales/app/controllers/categories_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    else
+      flash[:notice] = 'Category was not updated'
+      render :edit
+    end
+  end
+
+  def destroy
+    @category.destroy
+    flash[:notice] = 'Category was succesfully destroyed'
+    redirect_to companies_path
```
Comment (juliocabrera820) [inline local_sales/app/controllers/categories_controller.rb:39]: It must be categories_path

**Review by juliocabrera820 — DISMISSED** [review]
Nice job 👍🏼 ⚡

**Review by oramasangel — DISMISSED** [review]
LGTM

**Review by xFerchoVJ — APPROVED** [review]
LGTM!

File: local_sales/app/views/categories/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<h1>Categories#edit</h1>
```
Comment (jclopezdev) [inline local_sales/app/views/categories/edit.html.erb:1]: Hey, what do you think of updating these titles to some more presentable?

File: local_sales/app/views/categories/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<h1>Categories#index</h1>
```
Comment (jclopezdev) [inline local_sales/app/views/categories/index.html.erb:1]: Also here...

**Review by juliocabrera820 — APPROVED** [review]

File: local_sales/app/views/categories/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<%= form_with(model: category, local: true) do |form| %>
+
+  <div>
+    <label for="fname">Categoria:</label><br>
+    <%= form.text_field :name %><br>
```
Comment (oramasangel) [inline local_sales/app/views/categories/_form.html.erb:5]: you must disable the autocomplete

**Review by oramasangel — APPROVED** [review]
GJ

**Review by xFerchoVJ — APPROVED** [review]
LGTM

**Pull request merged**